xarray/core/variable.py

```diff
@@ -171,6 +174,14 @@ def _maybe_wrap_data(data):
     """
     if isinstance(data, pd.Index):
         return PandasIndexingAdapter(data)
+    if isinstance(data, pd.api.extensions.ExtensionArray) or (
+        isinstance(data, ExtensionDuckArray)
+        and isinstance(data.array, pd.api.extensions.ExtensionArray)
```

Comment:
```suggestion
```

This should not be needed, if it's wrapped it'll work with Variable

Reply:
Yes!